engine/consensus/compliance/engine.go

```diff
@@ -44,39 +45,45 @@ const defaultTimeoutObjectsQueueCapacity = 1000
 // and forwards outbound messages to the networking layer.
 // `compliance.Core` implements the actual compliance logic.
 type Engine struct {
-	unit                       *engine.Unit
-	lm                         *lifecycle.LifecycleManager
-	log                        zerolog.Logger
-	mempool                    module.MempoolMetrics
-	metrics                    module.EngineMetrics
-	me                         module.Local
-	headers                    storage.Headers
-	payloads                   storage.Payloads
-	tracer                     module.Tracer
-	state                      protocol.State
-	prov                       network.Engine
-	core                       *Core
-	pendingBlocks              engine.MessageStore
-	pendingRangeResponses      engine.MessageStore
-	pendingVotes               engine.MessageStore
-	pendingTimeouts            engine.MessageStore
-	messageHandler             *engine.MessageHandler
+	log      zerolog.Logger
+	mempool  module.MempoolMetrics
+	metrics  module.EngineMetrics
+	me       module.Local
+	headers  storage.Headers
+	payloads storage.Payloads
+	tracer   module.Tracer
+	state    protocol.State
+	prov     consensus.ProposalProvider
+	core     *Core
+	// queues for inbound messsages
+	pendingBlocks         engine.MessageStore
+	pendingRangeResponses engine.MessageStore
+	pendingVotes          engine.MessageStore
```

Comment:
we also need to get rid of `pendingVotes` and `pendingTimeouts`, shall we add a TODO at least?